### PR TITLE
[Merged by Bors] - feat: add some missing PowerSeries.coeff_C lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
These parallel the lemmas for Polynomial

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/HahnSeries.lean [hunk 1]
```diff
@@ -1270,12 +1270,9 @@ def toPowerSeriesAlg : HahnSeries ℕ A ≃ₐ[R] PowerSeries A :=
   { toPowerSeries with
     commutes' := fun r => by
       ext n
-      simp only [algebraMap_apply, PowerSeries.algebraMap_apply, C_apply,
-        coeff_toPowerSeries]
-      cases' n with n
-      · simp [PowerSeries.coeff_zero_eq_constantCoeff, single_coeff_same]
-      · simp [n.succ_ne_zero, Ne.def, not_false_iff, single_coeff_of_ne]
-        rw [PowerSeries.coeff_C, if_neg n.succ_ne_zero] }
+      simp [algebraMap_apply, PowerSeries.algebraMap_apply, PowerSeries.coeff_C, single_coeff,
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/HahnSeries.lean:1273]: Here
```
ext n
      cases n <;> simp [algebraMap_apply, PowerSeries.algebraMap_apply]
```
works, and I find it more readable than digging into `Decidable` instances... what do you think?

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

**Comment by riccardobrasca** [comment]
BTW `HahnSeries.lean` is a huge file... maybe we can add a TODO to split it.

File: Mathlib/RingTheory/HahnSeries.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/RingTheory/HahnSeries.lean]: Yeah. I'm a bit disappointed that we need to do the case-split, but avoiding it isn't as pretty as I'd hoped.

**Comment by Ruben-VandeVelde** [comment]
> BTW `HahnSeries.lean` is a huge file... maybe we can add a TODO to split it.

I didn't see an obvious splitting point, but happy to work on this if you do see one

**Pull request closed without merging**